File: Controllers/TestController.cs
using System.Collections;
using System.Collections.Generic;
using System.Threading.Tasks;
using KnowledgeControl.Services.Interfaces;
using KnowledgeControl.Models;
using Microsoft.AspNetCore.Mvc;

namespace KnowledgeControl.Controllers
{
    [ApiController]
    [Route("api/[controller]/[action]")]
    public class TestController : ControllerBase
    {
        private readonly ITestService _testService;

        public TestController(ITestService testService)
        {
            _testService = testService;
        }
        
        [HttpGet]
        public IEnumerable<TestViewModel> GetTests() =>
            _testService.GetTests();

        [HttpPost]
        public Task CreateTest(TestModel model) =>
            _testService.CreateTest(model);
        
        [HttpPut]
        public Task EditTest(TestModel model) =>
            _testService.EditTest(model);
        
        [HttpDelete]
        public Task DeleteTest(TestModel model) =>
            _testService.DeleteTest(model);
    }
}

File: ClientApp/src/api/testController.ts
import axios from "axios";

import {TestViewModel} from "../models/TestViewModel";
import {TestModel} from "../models/TestModel";

const getTests = (): Promise<TestViewModel[]> =>
  axios.get("/api/Tests")
    .then(response => response.data)

const getSolvedTests = (): Promise<TestViewModel[]> =>
  axios.get("/api/Tests/solved")
    .then(response => response.data)

const getTest = (id: number): Promise<TestModel> =>
  axios.get(`/api/Tests/${id}`)
    .then(response => response.data)

const createTest = (test: TestModel): Promise<TestModel> =>
  axios.post("/api/Tests", test)
    .then(response => response.data)

const editTest = (test: TestModel) =>
  axios.put("/api/Tests", test)

const deleteTest = (id: number) =>
  axios.delete(`/api/Tests/${id}`)

export default {
  getTests,
  getSolvedTests,
  getTest,
  createTest,
  editTest,
  deleteTest
}